run-build

```diff
@@ -95,11 +95,23 @@ class ImageBuildRoot(object):
         repo_path = config['ostree']['repodir']
         remote = config['ostree']['remote']
         ref = config['ostree']['ref']
+
+        # If we are on an OSTree system, use the system repo as an additional cache.
+        # In the case of an Endless OS developer system booted into the same or similar
+        # ref as we are building an image of, this saves pulling over the network. In
+        # other cases it is harmless.
+        system_ostree_repo = '/ostree/repo'
+        if os.path.isdir(system_ostree_repo):
+            extra_pull_args = ['--localcache-repo', system_ostree_repo]
+        else:
+            extra_pull_args = []
+
         log.info(f'Pulling OSTree ref {remote}:{ref}')
         eib.retry(subprocess.check_call, [
             'ostree',
             f'--repo={repo_path}',
             'pull',
+            *extra_pull_args,
```

Comment:
Does the list within the list get flattened? I wouldn't expect it, but maybe subprocess does that and I've never tried.

Reply:
That's what the `*` does:

```
>>> extra_pull_args = ['--localcache-repo', "/ostree/repo"]
>>> ["ostree", "pull", *extra_pull_args, "eos", "my-cool-ref"]
['ostree', 'pull', '--localcache-repo', '/ostree/repo', 'eos', 'my-cool-ref']
>>> ["ostree", "pull", extra_pull_args, "eos", "my-cool-ref"]
['ostree', 'pull', ['--localcache-repo', '/ostree/repo'], 'eos', 'my-cool-ref']
```

Reply:
Oh, I totally missed that. :+1:

Reply:
I had no idea you could compose lists that way. That's a nice feature.